### ISSUE TITLE
Fails to compile on Manjaro Linux with OpenSSL 1.1.1

### ISSUE THREAD

**Issue by K4rakara** [sfackler/rust-openssl#1321]
On Manjaro Linux with OpenSSL 1.1.1, this package fails to compile, with the following error:
```
  This crate is only compatible with OpenSSL 1.0.1, 1.0.2, and 1.1.0, or LibreSSL,
  but a different version of OpenSSL was found. The build is now aborting due to
  this version mismatch.
```

I tried following other peoples solutions of providing the environment variable `OPENSSL_DIR` and pointing it to the OpenSSL installation directory (`/usr/lib/openssl/` or similar, `/usr/lib/openssl-1.0/` for me?), but it just started throwing a different error:
```
  thread 'main' panicked at 'OpenSSL library directory does not exist: /usr/lib/openssl-1.0/lib', /home/jack/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.9.11/build.rs:56:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This confused me a lot, as there is no "lib" dir in OpenSSL for Linux.

I ended up getting the idea to try and install OpenSSL in Wine and see what the directory structure was, and welp, its got all three of the directories that it wanted. 

TL;DR: This package might be misidentifying Manjaro Linux as Windows.

**Comment by sfackler** [comment]
It appears that you are using an outdated copy of rust-openssl that does not support OpenSSL 1.1.1.

**Comment by K4rakara** [comment]
Huh, I guess you're right. By inspecting the `Cargo.lock`, it looks like `tungstenite` depends on `native-tls`, which in turn depends on `openssl 0.9.11`. I might be wrong, but I don't think there's a way to override the version of a dependency that a crate uses.

**Comment by sfackler** [comment]
The current release of tungstenite depends on a up-to-date version of native-tls.
